Assets/Mirror/Runtime/NetworkManager.cs

```diff
@@ -1348,17 +1345,19 @@ public Transform GetStartPosition()
         }
 
         /// <summary>
-        /// Called on the server when a client removes a player.
-        /// <para>The default implementation of this function destroys the corresponding player object.</para>
+        /// Called on the server when the player (identity) needs to be removed from the connection.
         /// </summary>
         /// <param name="conn">The connection to remove the player from.</param>
-        /// <param name="player">The player identity to remove.</param>
+        public virtual void OnServerRemovePlayer(NetworkConnection conn) { }
+
+        // Deprecated 04/25/2020
+        /// <summary>
+        /// Obsolete: Override <see cref="OnServerRemovePlayer(NetworkConnection conn)"/> instead.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never), Obsolete("Override OnServerRemovePlayer(NetworkConnection conn) instead")]
         public virtual void OnServerRemovePlayer(NetworkConnection conn, NetworkIdentity player)
         {
-            if (player.gameObject != null)
-            {
-                NetworkServer.Destroy(player.gameObject);
-            }
+            NetworkServer.Destroy(conn.identity.gameObject);
```

Comment:
if you are marking this function with Obsolete, you shouldn't also change the body of the function

Reply:
It doesn't change what happens at all, just removes a redundant null check.

Reply:
redundant? what if identity is null?

if identity is not ddol, and the scene changes for some reason then it will be null

Reply:
If the identity is not in DDOL Mirror will respawn it before anyone can send a RemovePlayer message.